ui/src/sandbox.rs

```diff
@@ -280,6 +281,27 @@ fn set_execution_environment(
     cmd.apply_backtrace(&req);
 }
 
+fn parse_magic_comments(code: &str) -> Vec<(&'static str, String)> {
+    lazy_static::lazy_static! {
+        pub static ref MIRIFLAGS: Regex = Regex::new(r#"(///|//!|//)\s*MIRIFLAGS\s*=\s*(.*)"#).unwrap();
```

Comment:
An outer attribute seems weird, here:

```suggestion
        pub static ref MIRIFLAGS: Regex = Regex::new(r#"(//!|//)\s*MIRIFLAGS\s*=\s*(.*)"#).unwrap();
```

Reply:
```rust
/// MIRIFLAGS=-Zmiri-tag-raw-pointers
fn main() {
}
```
seems plausible to me. My instinct here is to err on the side of accepting as much as possible, because I think the only alternative we have is silently ignoring things.

Reply:
I agree... whenever I use magic comments on ui tests I am nervous that they might be silently ignored.

Reply:
So, the thing is that:

```rs
#![feature(…)]

/// MIRIFLAGS=-Zmiri-tag-raw-pointers
fn main() {
```

would fail, IIUC, since it wouldn't be the first-line of a file. And I can imagine the transition from your snippet to mine occurring quite easily, especially given that

```rs
/// MIRIFLAGS=-Zmiri-tag-raw-pointers
#![feature(…)]

fn main() {
```

would not be valid Rust.

I'd rather have Miri start saying with which `MIRIFLAGS` it is being run (which seems like a nice addition overall), rather than have an outer attribute be accepter where an inner one should be used: it is all too reminiscent of:

```rs
//! src/lib.rs
#[forbid(unsafe_code)]

use …;

fn foo… /* actually allowed to use `unsafe`! */
```

Reply:
> would fail, IIUC, since it wouldn't be the first-line of a file.

Hm... good point. Playground will also add `feature` lines at the very top of the snippet when you click the error message that complains about a missing feature gate.

Reply:
I think `#![feature(...)]` is pretty problematic here. I was afraid magic comments would run into this kind of complexity. 

Parsing the magic flag on lines other than the first makes some sense in response to this, but then we run into this:
```rust
// This example works totally fine by default. But it breaks if we set
// MIRIFLAGS=-Zmiri-tag-raw-pointers
fn main() {
    🤦 
}
```

I agree that Miri should report its configuration somewhere.

Other things we could do...
1. Add a new UI element. I don't really want to do this because it's a lot of work, and also MIRIFLAGS is quite unique in that it needs to be provided as an environment variable when `cargo-miri` is run. All other build options we would want as arguments to `cargo rustc`.
2. Try to parse the first comment line without leading whitespace. Is it sufficiently unlikely to accidentally set MIRIFLAGS this way?

Reply:
I don't think this is unreasonable. It's not like people are shipping code using the playground, and this seems to be quite an edge case (https://grep.app/search?q=//%20MIRIFLAGS and https://cs.github.com/?scopeName=All+repos&scope=&q=%22%2F%2F+MIRIFLAGS%22 say enough of one that it doesn't exist in open source rust code).

Reply:
@thomcc Can you clarify what exactly "it" is here?
> I don't think this is unreasonable.

Reply:
> Parsing the magic comment on lines other than the first makes some sense in response to this, but then we run into this...

The issue you are describing here.

Reply:
I think it's fine if that turns on that MIRIFLAG